shoop/campaigns/models/basket_line_effects.py

```diff
@@ -9,15 +9,16 @@
 from django.db import models
 from django.utils.translation import ugettext_lazy as _
 
-from shoop.campaigns.models import BasketDiscountEffect
-from shoop.core.models import OrderLineType, Product
+from shoop.core.models import OrderLineType, PolymorphicShoopModel, Product
 from shoop.core.order_creator._source import LineSource
 
 
-class BasketLineEffect(BasketDiscountEffect):
+class BaseBasketLineEffect(PolymorphicShoopModel):
+    identifier = None
+    model = None
+    admin_form_class = None
 
-    class Meta:
-        abstract = True
+    campaign = models.ForeignKey("BasketCampaign", related_name='basket_line_effects', verbose_name=_("campaign"))
```

Comment:
It's a BasketCampaign so the word `basket` is a bit redundant here. The related name could be just `line_effects`.